### PR TITLE
handlePasswordGrant: insert connectorData into OfflineSession

### PR DESCRIPTION
**Pull request by hensur**
#### Overview

Using dex with the kubernetes oidc client results in the creation
of an Offline Session with empty ConnectorData.
This results in a working login, however, the ldap connector will fail once the client attempts to refresh its token.

This change will insert the ConnectorData from the initial Login
into the OfflineSession, as already done in `handlePasswordLogin`.

#### What this PR does / why we need it

Not doing this results in failures to refresh a token.
I've observed this with the ldap connetor, other connectors might
be affected as well.

The ldap connector will fail with this error message:
`failed to refresh identity: ldap: failed to unmarshal internal data: unexpected end of JSON input`

The `Refresh` method of the ldap connector will first try to unmarshal
the ConnectorData, which was empty, as dex wouldn't
store it in storage after a successful login.

#### Special notes for your reviewer

While this seems to fix the issue, I'm not sure if it is desired to save the ConnectorData in `handlePasswordGrant` as done in `handlePasswordLogin`.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by nabokihms** [comment]
Hello @hensur. Thanks, it looks good. I wonder if we can write a test for this?

**Comment by hensur** [comment]
Hey @nabokihms!
I had some time to look into this and came up with a test for this. It won't pass without my changes.
I took a lot of inspiration from the refresh handler tests, as I'm not really familiar with the dex code base :)

File: server/handlers_test.go [hunk 1]
```diff
@@ -227,3 +229,75 @@ func TestHandleAuthCode(t *testing.T) {
...
+"username": "test",
+"password": "test"
+}`),
+	}
+
+	err = s.CreateConnector(c1)
+	require.NoError(t, err)
+}
+
+func TestConnectorDataNotEmpty(t *testing.T) {
```
Comment (nabokihms) [inline server/handlers_test.go:259]: Let's give this test a more specific name like the `TestPasswordConnectorDataNotEmpty`.

**Review by nabokihms — APPROVED** [review]
Approve in advance. Besides the test function name, everything looks good to me.

File: server/handlers_test.go (hunk 1 above)
Comment (hensur) [reply server/handlers_test.go]: I renamed the test :)

**Comment by nabokihms** [comment]
Thanks for your contribution @hensur !

**Pull request merged**